### ISSUE TITLE
Same pre-optimization and post-optimization channel budget allocation ratios , but suggesting much higher budget instead of aligning the budget to the one i requested.

### ISSUE THREAD

**Issue by entzyeung** [google/lightweight_mmm#297]
Dear LMMM Team,

I want to express my sincere appreciation for the hard work you've dedicated to creating this remarkable model. It's truly impressive and has proven to be user-friendly and easy to install, unlike my previous attempt with the FB Robyn model, which I eventually abandoned due to its complexity. Discovering your model felt like a ray of sunshine after a cloudy day.

To cut to the chase, I've encountered a similar issue to that of @DongHarry-Kang. I've spent considerable time experimenting with various datasets and fine-tuning for two weeks. However, I consistently obtain similar results when calling the `plot_pre_post_budget_allocation_comparison` function.

I've included some sample outputs below:

![Output 1](https://github.com/google/lightweight_mmm/assets/60934147/398725ef-c337-4f6c-9599-0dd14218ca54)

![Output 2](https://github.com/google/lightweight_mmm/assets/60934147/eb904362-c22d-49bd-9406-b1f8a76e1e64)

I've thoroughly examined data quality using `check_data_quality()`, assessed variances with `highlight_variances()`, reviewed spend fractions with `highlight_low_spend_fractions()`, and analyzed variance inflation factors with `highlight_high_vif_values()`. The datasets appear to be in good shape and suitable for analysis. I even trained the best model with parameters obtained from Gridsearches. However, when it comes to the critical aspect of budget allocation, it seems there may be an issue.

Regardless of how I adjust the budget and cost parameters, the model consistently produces the same outcome: maintaining the same channel ratios while suggesting a higher budget allocation.

Allow me to illustrate this with an example.
<img width="705" alt="Screenshot 2024-01-07 at 22 55 23" src="https://github.com/google/lightweight_mmm/assets/60934147/77b11f6d-173e-4a19-9d53-ac24aba14dec">

As you can see from the investigation, the model did suggest me different and higher budget while maintaining the same channel ratios. I am so confused that why the model not reallocating the budgets as the documentation said? and keep increasing our budget instead?
is there any params I missed? if not, is there any way to fix it?

Here is my syntax when calling the optimizatin function:
# Run optimization with the parameters of choice.
```
solution, kpi_without_optim, previous_media_allocation = optimize_media.find_optimal_budgets(
            n_time_periods= n_time_periods,
            media_mix_model= mmm,
            extra_features= extra_features_test[:n_time_periods],# it has been scaled.
            budget= budget,
            prices= prices,
            media_scaler = media_scaler,
            target_scaler = target_scaler,
            )
```

# Plot out pre post optimization budget allocation and predicted target variable comparison.
```
plot.plot_pre_post_budget_allocation_comparison(media_mix_model=mmm, 
                                                kpi_with_optim=solution['fun'], 
                                                kpi_without_optim=kpi_without_optim,
                                                optimal_buget_allocation=optimal_buget_allocation, 
                                                previous_budget_allocation=previous_budget_allocation, 
                                                figure_size=(10,10))
```
Thank you in advance.

**Comment by santiarias** [comment]
@entzyeung could you post your baseline_contribution_area plot?

**Comment by entzyeung** [comment]
Thank you @santiarias, here you go.
<img width="1504" alt="Screenshot 2024-01-12 at 15 25 48" src="https://github.com/google/lightweight_mmm/assets/60934147/4aebf6ed-18c5-41d7-854d-a629be7c6366">

and i have another discussion going on: [here](https://github.com/google/lightweight_mmm/issues/294#issuecomment-1887982811)

**Comment by entzyeung** [comment]
No problem at all, but I m outside right now, I will send once I go back home.

**Comment by entzyeung** [comment]
@santiarias here you go!

<img width="836" alt="Screenshot 2024-01-12 at 21 09 59" src="https://github.com/google/lightweight_mmm/assets/60934147/fea909e3-be23-4a99-976a-c6f770811ea9">
<img width="577" alt="Screenshot 2024-01-12 at 21 10 22" src="https://github.com/google/lightweight_mmm/assets/60934147/4d612e82-0b0e-455e-a649-d6cba2aeec8b">

**Comment by entzyeung** [comment]
@santiarias please let me know if you want some extra info about my model anytime.

**Comment by santiarias** [comment]
@entzyeung yeah if you could just show a very small sample of your training data (just a few rows is enough)

**Comment by entzyeung** [comment]
@santiarias sure thing, here you go.
[Data](https://drive.google.com/file/d/1juMcn-hOPNaea1U0kdMYMJ6EjfHfCeeh/view?usp=drive_link)
I extracted the first 10 rows of each dataset just before applying jnp.array to them.
<img width="814" alt="Screenshot 2024-01-16 at 18 20 02" src="https://github.com/google/lightweight_mmm/assets/60934147/c6247729-dea7-4043-8f17-c3ac6596e400">

**Comment by entzyeung** [comment]
@santiarias does the link work? please feel free to let me know if it is not working.

**Comment by entzyeung** [comment]
Would the development team help?

**Comment by aviv-ianphilipp** [comment]
It's just a hunch but I have a feeling its because the prices are order of magnitudes off. Are your prices in the optimization costs per thousand or costs per unit. Is the model designed for costs per thousand?

**Comment by KRMAR12** [comment]
@entzyeung 

The model looks very good. For some reason I can't open the data file

**Comment by KRMAR12** [comment]
I really need your help with this. Can you send me your data set?
@entzyeung
